CMakeLists.txt

```
@@ -54,43 +41,65 @@ add_library(${PROJECT_NAME}
   src/imarker_robot_state.cpp
   src/imarker_end_effector.cpp
 )
-set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
-add_dependencies(${PROJECT_NAME} graph_msgs_generate_messages_cpp)
-target_link_libraries(${PROJECT_NAME}
-  ${catkin_LIBRARIES} ${Boost_LIBRARIES}
+
+target_link_libraries(
+  ${PROJECT_NAME}
+)
+target_include_directories(${PROJECT_NAME}
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  PUBLIC $<INSTALL_INTERFACE:include>
 )
+ament_target_dependencies(${PROJECT_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
 # Demo executable
-add_executable(${PROJECT_NAME}_demo src/${PROJECT_NAME}_demo.cpp)
+add_executable(${PROJECT_NAME}_demo
+  src/${PROJECT_NAME}_demo.cpp
+)
 target_link_libraries(${PROJECT_NAME}_demo
-  ${catkin_LIBRARIES} ${PROJECT_NAME}
+  ${PROJECT_NAME}
 )
+ament_target_dependencies(${PROJECT_NAME}_demo ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
 
-#############
-## Testing ##
-#############
-
-## Test for correct C++ source code
-roslint_cpp()
+# Exports
+ament_export_targets(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
+ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
 
 #############
 ## Install ##
 #############
 
 # Install libraries
-install(TARGETS ${PROJECT_NAME}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+install(
+  TARGETS
+    ${PROJECT_NAME}
+  EXPORT
+    export_${PROJECT_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION lib/${PROJECT_NAME}
+)
 
 # Install header files
-install(DIRECTORY include/${PROJECT_NAME}/   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+install(DIRECTORY include DESTINATION include)
```

I am using your branch here on another project, and `#include <moveit_visual_tools/moveit_visual_tools.h>` was failing. On further inspection, `moveit_visual_tools` includes were being installed to `install/moveit_visual_tools/include/include/moveit_visual_tools/` (notice the extra include)

I am not sure exactly why, but adding a `/` during the install command seems to fix this problem, removing the extra `include` in the path, and allowing the other package to find the file

```suggestion
install(DIRECTORY include/ DESTINATION include)
```